src/langchain_google_spanner/vector_store.py

```diff
@@ -0,0 +1,1129 @@
+            query_parameter (QueryParameters): The query parameters. Defaults to QueryParameters().
+
+        Returns:
+            SpannerVectorStore: Initialized SpannerVectorStore instance.
+        """
+        texts = [d.page_content for d in documents]
+        metadatas = [d.metadata for d in documents]
+        return cls.from_texts(
+            texts,
+            embedding,
+            metadatas=metadatas,
+            embedding_service=embedding,
+            id_column=id_column,
+            content_column=content_column,
+            embedding_column=embedding_column,
+            client=client,
+            ids=ids,
+            metadata_columns=metadata_columns,
+            ignore_metadata_columns=ignore_metadata_columns,
+            metadata_json_column=metadata_json_column,
+            query_parameter=query_parameter,
+            kwargs=kwargs,
+        )
+
+    @classmethod
+    def from_texts(
+        cls: Type[SpannerVectorStore],
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        id_column: str = ID_COLUMN_NAME,
+        content_column: str = CONTENT_COLUMN_NAME,
+        embedding_column: str = EMBEDDING_COLUMN_NAME,
+        ids: Optional[List[str]] = None,
+        client: Client = Client(),
+        metadata_columns: Optional[List[str]] = None,
+        ignore_metadata_columns: Optional[List[str]] = None,
+        metadata_json_column: Optional[str] = None,
+        query_parameter: QueryParameters = QueryParameters(),
+        **kwargs: Any,
+    ) -> SpannerVectorStore:
+        """
+        Initialize SpannerVectorStore from a list of texts.
+
+        Args:
+            texts (List[str]): List of texts.
+            embedding (Embeddings): The embedding service.
+            metadatas (Optional[List[dict]]): Optional list of metadatas associated with the texts. Defaults to None.
+            id_column (str): The name of the row ID column. Defaults to ID_COLUMN_NAME.
+            content_column (str): The name of the content column. Defaults to CONTENT_COLUMN_NAME.
+            embedding_column (str): The name of the embedding column. Defaults to EMBEDDING_COLUMN_NAME.
+            ids (Optional[List[str]]): Optional list of IDs for the texts. Defaults to None.
+            client (Client): The Spanner client. Defaults to Client().
+            metadata_columns (Optional[List[str]]): List of metadata columns. Defaults to None.
+            ignore_metadata_columns (Optional[List[str]]): List of metadata columns to ignore. Defaults to None.
+            metadata_json_column (Optional[str]): The generic metadata column. Defaults to None.
+            query_parameter (QueryParameters): The query parameters. Defaults to QueryParameters().
+
+        Returns:
+            SpannerVectorStore: Initialized SpannerVectorStore instance.
+        """
+        instance_id = get_from_dict_or_env(
```

Comment:
Don't use kwargs and add use comment `# type: ignore[override]` to ignore the warning

Reply:
In my opinion, we need kwards to dervide the value if user is passing a dictionary and we can't take these values explicitly in the method signature because of interface constraints.

Reply:
The other implementations are ignoring the interface constraint. I don't think we should support env vars or kwargs

Reply:
It should be safe to override the interface and add the additional required arguments for instance_id, database_id and table_name (see https://github.com/googleapis/langchain-google-cloud-sql-pg-python/blob/main/src/langchain_google_cloud_sql_pg/vectorstore.py#L390). The developer experience is not good when it errors on missing kwards or needing to set env vars.